accessibility_api/AccessibilityRequestHandler.py

```diff
@@ -0,0 +1,127 @@
+'''
+This Source Code Form is subject to the terms of the Mozilla Public
+License, v. 2.0. If a copy of the MPL was not distributed with this file,
+You can obtain one at http://mozilla.org/MPL/2.0/.
+'''
+
+import sys
+from BaseHTTPServer import BaseHTTPRequestHandler
+from urlparse import urlparse, parse_qs, parse_qsl
+from json import dumps
+from comtypes import CoInitialize
+from accessibility_api.accessibility_lib.scripts.constants import (
+    SUCCESSFUL_RESPONSE, ERROR_RESPONSE
+)
+from accessibility_api.accessibility_lib.scripts.accessible import accessible
+from accessibility_api.accessibility_lib.scripts.event import event
+from accessibility_api.accessibility_lib.scripts.commands import (
+    execute_command
+)
+
+
+class AccessibilityRequestHandler(BaseHTTPRequestHandler):
+    """ Service requests for accessible objects """
+    def _successful_response(self, json):
+        self.send_response(200)
+        self.send_header('Content-Type', 'application/json')
+        self.end_headers()
+        self.wfile.write(json)
+
+    def _bad_request(self, message):
+        self.send_response(500, message)
+        self.send_header('Content-Type', 'application/json')
+        self.end_headers()
+
+    def do_GET(self):
+        '''Proccess get requests'''
+        url = urlparse(self.path)
+        handler = getattr(self, "get_" + url.path[1:], None)
+        if handler:
+            handler(url.query)
+        else:
+            self.send_response(403, 'Invalid Request')
+            self.send_header('Content-Type', 'application/json')
+            self.end_headers()
+
+        return
+
+    def response(self, message, key, value, status):
+        """Returns appropriate response"""
```

Comment:
Ah ok, just looked through the stuff, so it is almost as what I was thinking. So the idea is to have the following JSON reposnse (on the client side):

{
  result: ...,
  error: ...
}
So then the client will simply check for response.error and if there's a falsy value there, the client will just go for the result.

In terms of you internal logic, it took me some time to see how you process these things. It would likely be more readable to set error and result values as soon as possible as early as serialize_result in Accessible or a similar location. Then as you move up the stack you just check if error do that if not go for the result. This way we eliminate the status based lookup.

^ Granted that I understood the code correctly :)